src/input/mod.rs

```diff
@@ -235,128 +235,148 @@ impl Input {
         });
     }
 
-    pub fn button(
-        &self,
-        button: Button,
-    ) -> ButtonInfo {
+    pub fn hit<H: Hit>(&self, hit: H) -> bool {
+        hit.hit(self)
+    }
+
+    pub fn delta<D: Delta>(&self, delta: D) -> <D as Delta>::Output {
+        delta.delta(self)
+    }
+
+    pub fn timed_delta<D: Delta>(&self, delta: D) -> Option<TimerDuration> {
+        delta.timed_delta(self)
+    }
+
+    pub fn hit_count<C: HitCount>(&self, hit_count: C) -> <C as HitCount>::Output {
+        hit_count.hit_count(self)
+    }
+}
+
+/// Keyboard or mouse button.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum Button {
+    /// Keyboard button.
+    Key(Key),
+    /// Mouse button.
+    Mouse(MouseButton),
+}
+
+pub trait Hit {
+    fn hit(&self, input: &Input) -> bool;
+}
+
+impl Hit for Button {
+    fn hit(&self, input: &Input) -> bool {
+        match *self {
+            Button::Key(button) => input.state.keys_pressed.contains(&button),
+            Button::Mouse(button) => input.state.mouse_pressed.contains(&button),
+        }
+    }
+}
+
+impl Hit for axis::Key {
+    fn hit(&self, input: &Input) -> bool {
+        input.delta.keys_hit.iter().filter(|&&k| k == self.pos || k == self.neg).count() > 0
+    }
+}
+
+impl Hit for axis::Raw {
+    fn hit(&self, input: &Input) -> bool {
+        input.delta.axes_raw.iter().filter(|&&(id, _)| id == self.id).count() > 0
+    }
+}
+
+pub trait HitCount {
+    type Output;
```

Comment:
Surprised to see an associated type here.

Reply:
That's because of `axis::Key`

Reply:
Oh, never mind... I just saw `type Output = (u8, u8)`.

Reply:
Not a big deal in my opinion